frontend/packages/core/src/Input/radio-group.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import {
+  FormControl as MuiFormControl,
+  FormControlLabel,
+  FormLabel as MuiFormLabel,
+  Radio,
+  RadioGroup as MuiRadioGroup,
+} from "@material-ui/core";
+import styled from "styled-components";
+
+const FormLabel = styled(MuiFormLabel)`
+  ${({ theme }) => `
+  && {
+    color: ${theme.palette.text.primary};
+    font-weight: bold;
+  }
+  `}
+`;
+
+const StyledRadioGroup = styled(MuiRadioGroup)`
+  ${({ ...props }) => `
+  display: flex;
+  width: 100%;
+  max-width: ${props["data-max-width"] || "500px"};
+  `}
+`;
+
+const FormControl = styled(MuiFormControl)`
+  ${({ theme, ...props }) => `
+  display: flex;
+  width: 100%;
+  margin: 16px 0;
+  max-width: ${props["data-max-width"] || "500px"};
+  .MuiInput-underline:after {
+    border-bottom: 2px solid ${theme.palette.accent.main};
+  }
+  `}
+`;
+
+interface RadioGroupOption {
+  label: string;
+  value?: string;
+}
+
+interface RadioGroupProps {
+  defaultOption?: number;
+  label?: string;
+  maxWidth?: string;
+  name: string;
+  options: RadioGroupOption[];
+  onChange: (value: string) => void;
+}
+
+const RadioGroup: React.FC<RadioGroupProps> = ({
+  defaultOption = 0,
+  label,
+  maxWidth,
+  name,
+  options,
+  onChange,
+}) => {
+  if (options.length === 0) {
+    return null;
+  }
+
+  const defaultIdx = defaultOption < options.length ? defaultOption : 0;
+  const [selectedIdx, setSelectedIdx] = React.useState(defaultIdx);
+
+  const updateSelectedOption = (event: React.ChangeEvent<{ name?: string; value: string }>) => {
+    const { value } = event.target;
+    const optionValues = options.map(o => o.value || o.label);
+    setSelectedIdx(optionValues.indexOf(value));
+    if (onChange !== undefined) {
+      onChange(value);
+    }
+  };
+
+  React.useEffect(() => {
+    if (onChange !== undefined) {
+      onChange(options[selectedIdx].value || options[selectedIdx].label);
+    }
+  }, []);
+
+  return (
+    <FormControl key={name} data-max-width={maxWidth}>
+      {label && <FormLabel>{label}</FormLabel>}
+      <StyledRadioGroup
+        aria-label={label}
+        name={name}
+        defaultValue={options[defaultIdx].value || options[defaultIdx].label}
+        onChange={updateSelectedOption}
+      >
+        {options.map(option => {
+          return (
+            <FormControlLabel
+              key={option.label}
+              value={option.value || option.label}
+              control={<Radio />}
```

Comment:
Might be cool to use the accent color for the selected option:
```tsx
const Radio = styled(MuiRadio)`
  ${({ theme }) => `
  &.Mui-checked {
    color: ${theme.palette.accent.main};
  }
  `}
`;

```

Reply:
I like it
![ezgif com-gif-maker (18)](https://user-images.githubusercontent.com/4410346/97025660-3dc4d500-150d-11eb-8ffd-69fd6d7b4d5b.gif)